R/impute_missing_data.R

```diff
@@ -0,0 +1,127 @@
+#' Impute missing data in a data frame
+#'
+#' This function imputes missing data in a data frame using one or more specified methods.
+#' The function can impute missing data for all columns or for a selected set of columns.
+#'
+#' @param df A data frame containing the data to impute.
+#' @param cols_list A list of character vectors specifying the columns to impute for each method.
+#'   Each element of the list should be a character vector containing the names of the columns to impute
+#'   using the corresponding method in the `methods` parameter. The default value is a list containing all column names of `df`.
+#' @param methods A character vector specifying the methods to use for imputing missing data.
+#'  The available methods are `"mean"`, `"median"` and `"mode"`. The default value is `"mean"`.
+#'
+#' @return A data frame with the same dimensions as `df` where missing values have been imputed using the specified methods.
+#'
+#' @examples
+#' df <- data.frame(x <- c(1, NA, 3, NA),y <- c(NA, 4, 6, 8))
+#' impute_missing_data(df)
+#'
+#' @export
+#'
+impute_missing_data <- function(df, cols_list=list(colnames(df)), methods=c("mean")) {
+  impute_data <- function(i) {

```

Comment:
Sprawdzenie poprawności argumentów (is.data.frame(df), methods %in% c('mean', 'median', 'mode', itp.)